js/memory.js

```diff
@@ -1,9 +1,43 @@
+console.log('memory.js connected');
+
 class MemoryGame {
-  constructor(cards){
+  constructor(cards) {
     this.cards = cards;
-    // add the rest of the class properties here
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
+  }
+
+  // Shuffle basado en el Fisher-Yates Shuffle algorithm
+  shuffleCards() {
+    let currentIndex = this.cards.length;
+    let temporaryValue;
+    let randomIndex;
+    while (0 !== currentIndex) {
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex -= 1;
+      temporaryValue = this.cards[currentIndex];
+      this.cards[currentIndex] = this.cards[randomIndex];
+      this.cards[randomIndex] = temporaryValue;
+    }
+  }
+
+  // Checkeamos si las cards son iguales
+  checkIfPair(card1, card2) {
+    this.pairsClicked++;
+    if (card1 === card2) {
+      this.pairsGuessed++;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  // Checkeamos si el user ha ganado
```

Comment:
Acordate que podes devolver directamente

`return this.pairsGuessed === this.cards.length / 2`

Reply:
Ah tienes razón! En un inicio lo probé con el ternary operator, pero Jasmine me devolvía error, así que decidí dejarlo con el if/else. Ahora lo he checkeado como dices y he visto que error era porque me dejaba el return 😅